### DISCUSSION TITLE
What is the main proper of the ErrorSubject class?

### DISCUSSION THREAD

**Discussion by nguyenvantruong17** [gialynguyen/vcc-schema#9] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/18642063/120356198-434d8d80-c32e-11eb-8990-11896bab889f.png)

Mục đích chính của class này là để làm gì?

**Comment by nguyenvantruong17** [comment]
prerequisite property dành cho mục đích gì nhỉ?

**Reply by gialynguyen** [reply]
sẽ dừng quá trình parse khi gặp một lỗi là prerequisite. Ví dụ:

`
 const name = string().min(4);
 name.parser(5)
`
sẽ không chạy đến hàm min(4), vì string() đã throw 1 lỗi prerequisite (cái lỗi liên quan đến data type thường là prerequisite)

**Comment by gialynguyen** [comment, marked as answer]
là 1 đơn vị Error được sinh ra quá trình parse diễn ra.
